### ISSUE TITLE
app crashes on message notification clicked

### ISSUE THREAD

**Issue by hkpanchani** [zulip/zulip-mobile#4484]
Login to multiple account lets say account 1 and account 2
Currently account 1 is active and you receive message notification on account 2.
On click of that notification app crashes and shows blank(white) screen.
And in pre-release version 27.158 it totally crashes and goes to loop of that zulip loading splash screen.
in pre-release version it doesn't even get recovered by force closing and clearing cache.. I have to keep clearing app data and add both account again.

**Comment by hkpanchani** [comment]
This is is already in working since 2018 here [](https://github.com/zulip/zulip-mobile/issues/2295)

**Comment by Gautam-Arora24** [comment]
@hkpanchani I have faced a similar type of issue before. I have just created a dummy account a few days ago to test the multiple account scenario in Zulip and faced this issue. May be @gnprice and @chrisbobbe can help us with it.

**Comment by gnprice** [comment]
Thanks @hkpanchani for the report! That's a bad symptom -- this is definitely something we'll want to fix.

We have #2295 already, but the symptoms that used to have (described [here](https://github.com/zulip/zulip-mobile/issues/2295#issuecomment-548612532)) were quite a bit milder than this.

**Comment by gnprice** [comment]
@hkpanchani and @Gautam-Arora24: when you see this issue happening, do you have only those two accounts listed in the app? Or do you have some other accounts too?

There's a known issue described at https://github.com/zulip/zulip-mobile/issues/4309#issuecomment-750446007 which can have a symptom like this one if you have e.g. account 1, account 2, and account 3, and accounts 2 and 3 are on the same organization/realm as each other.

**Comment by Gautam-Arora24** [comment]
Yes, @gnprice I face this issue only with two accounts listed in the app.

**Comment by hkpanchani** [comment]
Yes @gnprice i have tried with two accounts and 3 also same problem whrn switching accounts from notification..

**Comment by chrisbobbe** [comment]
@Gautam-Arora24, when you close the app and re-open it after the crash, do you have the same symptom of getting stuck on the loading screen and have to clear the app's data?

@hkpanchani, can you find out what the Zulip server version numbers are for the servers you're connecting to? For example, for chat.zulip.org, I can visit https://chat.zulip.org/api/v1/server_settings, and search for "zulip_version" there. You'd want to use the actual realm URL for the part before `/api/v1/server_settings` though, instead of chat.zulip.org (unless of course this is happening on chat.zulip.org 🙂).

**Comment by hkpanchani** [comment]
@chrisbobbe I was on zulip release version 3.3. Due to this problem i upgraded my server to git master("4.0-dev-2441-ge971cff5b8"). In both cases same happens.. 

And yes same case on chat.zulip.org..
I am active on chat.zulip.org and received a chat from another i open app crashes..

**Comment by Gautam-Arora24** [comment]
> when you close the app and re-open it after the crash, do you also have the symptom of getting stuck on the loading screen and you have to clear the app's data?

No Chris. The app starts working fine when I reopen it. Though I am seeing another issue. When I click on the notification ideally it should open another account where I got the message but instead, I am being redirected to the same account's message thread.

To explain you more clearly ->
Suppose I have A and B accounts on my Zulip app. I am currently logged into A. A sends a message to B. So a notification gets popped up on the phone stating that A has sent you a message. When I click on that notification, instead of showing the message in B's account. I am being redirected to A's account, therefore, showing the same thread.

**Comment by chrisbobbe** [comment]
> The app starts working fine when I reopen it. Though I am seeing another issue. When I click on the notification ideally it should open another account where I got the message but instead, I am being redirected to the same account's message thread.

Thanks for that detail, @Gautam-Arora24! This looks like #4631. Let's focus the conversation for that issue there, to give more room here for the issue that @hkpanchani reported.

**Comment by chrisbobbe** [comment]
@hkpanchani, I'd be interested to know if this issue might have been fixed (or its symptoms improved, anyway) by some recent releases.

I have a hypothesis that at least part of the issue has been fixed: 

- [ ] If you update the app to the beta v27.162 release (or later), are you still forced to clear app data and add both accounts again?

#4588 made the app more robust to the data in storage being corrupted because of a crash. (#4694 is open to make it less likely for the data to get corrupted in the first place, and we plan more work in that direction after that.)

You've [mentioned](https://github.com/zulip/zulip-mobile/issues/4484#issuecomment-785734666) that the issue does present itself when you have just two accounts logged in, so it seems like the particular [piece](https://github.com/zulip/zulip-mobile/issues/4309#issuecomment-750446007) of #4309 that Greg mentioned isn't involved. If #4309 is involved at all, I'm not quite sure how it would be, but I think there's a way to test if it's involved. Greg [says](https://github.com/zulip/zulip-mobile/issues/4309#issuecomment-750516036),

> "I believe this is now fixed for PMs […] Still open for notifications for stream messages"

So,

- [ ] Do you find a difference in behavior when the notification you tap is for a PM message vs. a stream message?

Ah, and another thing, with this in mind:

> app crashes and shows blank(white) screen

#4585, also in the v27.162 beta release, aims to improve the experience of this kind of crash, by giving a stack trace and a button to copy it to the clipboard.

- [ ] Do you see an error screen with a stack trace when you experience the issue? If so, it'd be great if you could paste it here (formatted with GitHub's [``` syntax for fenced code blocks](https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks#fenced-code-blocks).

**Comment by matthewlswanson** [comment]
I have this happening on the beta but with a single account only. I tap the new message notification and nothing happens but the notification goes away. I left beta program and reinstalled and that fixed it. On Samsung Galaxy S10+, latest build

**Comment by WesleyAC** [comment]
@matthewlswanson Thanks for reporting this! That's #4818, which is only in beta, and we should have a fix out for soon.
